akka-persistence-typed/src/main/scala/akka/persistence/typed/dstate/scaladsl/DurableStateBehavior.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.dstate.scaladsl
+
+import akka.actor.typed.internal.BehaviorImpl.DeferredBehavior
+import akka.actor.typed.internal.LoggerClass
+import akka.actor.typed.{BackoffSupervisorStrategy, Signal}
+import akka.persistence.typed.PersistenceId
+import akka.persistence.typed.dstate.internal.DurableStateBehaviorImpl
+
+object DurableStateBehavior {
+
+  type CommandHandler[Command, State] = (State, Command) => Effect[State]
+
+  private val logPrefixSkipList = classOf[DurableStateBehavior[_, _]].getName :: Nil
+
+  def apply[Command, State](
+      persistenceId: PersistenceId,
+      emptyState: State,
+      commandHandler: CommandHandler[Command, State]): DurableStateBehavior[Command, State] = {
+    val loggerClass = LoggerClass.detectLoggerClassFromStack(classOf[DurableStateBehavior[_, _]], logPrefixSkipList)
+    DurableStateBehaviorImpl(persistenceId, emptyState, commandHandler, loggerClass)
+  }
+
+}
+
+trait DurableStateBehavior[Command, State] extends DeferredBehavior[Command] {
```

Comment:
Is this intended to be use-by-extending API? For the Scala API side of `EventSourcedBehavior` we have it marked as "not for user extension" and users only use the factory style. Always nice to avoid inheritance-based APIs where possible (mostly because it makes bin comp hard).

Reply:
I just copied the same style as for `EventSourcedBehavior`, but I may have failed to copy one or other annotation. In any case, I was more interested in exercising the shape of the API. We will go through all those details later on.